### PR TITLE
Refactor QFN 3d element and added QFN Footprinter3d example

### PR DESCRIPTION
**Pull request by anas-sarkez**
Screenshot 2024-11-15 at 2 50 45 PM

### CODE REVIEW

**Review by abhijitxy — APPROVED** [review]

File: examples/qfn.example.tsx [hunk 1]
```diff
@@ -1,12 +1,24 @@
 import { JsCadFixture } from "jscad-fiber"
 import { QFN } from "../lib/qfn"
 import { ExtrudedPads } from "../lib/ExtrudedPads"
+import { fp } from "@tscircuit/footprinter"
 
 export default () => {
+  const footprint = "qfn24_w8_h8_thermalpad3.2mmx3.2mm"
+  const fpJson = fp.string(footprint).json()
+  console.log(fpJson)
```
Comment (seveibar) [inline examples/qfn.example.tsx:9]:
```suggestion
```

File: examples/qfn.example.tsx [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+        bodyWidth={8}
+        bodyLength={8}
+        thermalPadSize={{
+          width: 3.2,
+          length: 3.2,
+        }}
+      />
+      <ExtrudedPads footprint={footprint} />
     </JsCadFixture>
   )
```
Comment (seveibar) [inline examples/qfn.example.tsx:23]: Can you add a test that uses FootprinterModel

Comment (seveibar) [reply examples/qfn.example.tsx:23]: Or sorry, Footprinter3d

**Review by seveibar — CHANGES_REQUESTED** [review]
Nice work, minor changes requested

File: examples/qfn.example.tsx (hunk 2 above)
Comment (anas-sarkez) [reply examples/qfn.example.tsx:23]: @seveibar 
already added
examples/footprinter3d/fp-qfn56.example.tsx

**Review by seveibar — APPROVED** [review]

**Pull request merged**